providers/hp/ilo/helpers.go

```diff
@@ -146,6 +160,28 @@ func (i *Ilo) cmpAccessSettings(cfg *cfgresources.Network) (AccessSettings, bool
 	currentConfig.IpmiPort = cfg.IpmiPort
 	currentConfig.SSHStatus = sshEnable
 	currentConfig.SSHPort = cfg.SSHPort
+	snmpSettings := new(SNMPSettings)
+	snmpSettings.SnmpPort = 161 // TODO: Change this to something user-configurable
+	snmpSettings.TrapPort = 162 // TODO: Change this to something user-configurable
+	switch i.HardwareType() {
```

Comment:
same as my previous suggestion 
- implement a method to return the int pointer with the value set
- use the ilo4/ilo5 [constants](https://github.com/bmc-toolbox/bmclib/blob/58619e55ffc178bd531f9cda88fd9a328edf05dc/providers/hp/ilo/ilo.go#L33) in the case conditions

Reply:
I don't understand "return the int pointer with the value set".

Do you want an auxiliary function that creates the int pointer depending on hardware type and config, and just do something like

`currentConfig.SNMPSettings = setSnmpSettings(hardwareType, currentConfig)` ?

Reply:
yep 👍